packages/flutter_tools/bin/macos_build_flutter_assets.sh

```diff
@@ -40,6 +40,37 @@ if [[ -n "$TRACK_WIDGET_CREATION" ]]; then
   track_widget_creation_flag="--track-widget-creation"
 fi
 
+# Copy the framework and handle local engine builds.
+derived_dir="${SOURCE_ROOT}/../build/macos/Build/Products/flutter_framework"
+framework_path="${FLUTTER_ROOT}/bin/cache/artifacts/engine/darwin-x64"
+flutter_framework="${framework_path}/FlutterMacOS.framework"
```

Comment:
The framework name is used a few times; let's make it a variable.

Reply:
Done